embedded-graphics/src/draw_target.rs

```diff
@@ -0,0 +1,322 @@
+///
+/// `DrawTarget` also provides a way to define accelerated drawing methods for graphical primitives
+/// such as lines, rectangles, triangles, and circles.
+///
+/// Note that some displays require a "flush" operation to write changes from a framebuffer to the
+/// display. See docs associated with the chosen display driver for details on how to update the
+/// display.
+///
+/// # Examples
+///
+/// ## Implement `DrawTarget` for an 8 bit grayscale display
+///
+/// This example uses an imaginary display that has a 64x64px framebuffer of 8 bit values that
+/// is sent to the display over a (simplified) SPI interface.
+///
+/// ```rust
+/// use embedded_graphics::{
+///     drawable::Pixel,
+///     egcircle,
+///     geometry::Size,
+///     pixelcolor::{Gray8, GrayColor},
+///     prelude::*,
+///     primitive_style, DrawTarget,
+/// };
+///
+/// # struct SPI1;
+/// #
+/// # impl SPI1 {
+/// #     pub fn send_bytes(&self, buf: &[u8]) -> Result<(), ()> {
+/// #         Ok(())
+/// #     }
+/// # }
+/// #
+/// /// A fake display 64px x 64px where each pixel is stored as a single `u8`
+/// struct ExampleDisplay {
+///     framebuffer: [u8; 64 * 64],
+///     iface: SPI1,
+/// }
+///
+/// impl ExampleDisplay {
+///     /// Send buffer to the display
+///     pub fn flush(&self) -> Result<(), ()> {
+///         self.iface.send_bytes(&self.framebuffer)
+///     }
+/// }
+///
+/// impl DrawTarget<Gray8> for ExampleDisplay {
+///     type Error = core::convert::Infallible;
+///
+///     /// Draw a `Pixel` that has a color defined as `Gray8`.
+///     fn draw_pixel(&mut self, pixel: Pixel<Gray8>) -> Result<(), Self::Error> {
+///         let Pixel(coord, color) = pixel;
+///         // Place an (x, y) pixel at the right index in the framebuffer
+///         let index = coord.x + coord.y * 64;
+///         self.framebuffer[index as usize] = color.luma();
+///
+///         Ok(())
+///     }
+///
+///     fn size(&self) -> Size {
+///         Size::new(64, 64)
+///     }
+/// }
+///
+/// let mut display = ExampleDisplay {
+///     framebuffer: [0; 4096],
+///     iface: SPI1,
+/// };
+///
+/// // Draw a circle centered around `(32, 32)` with a radius of `10` and a white stroke
+/// let circle = egcircle!(
+///     center = (32, 32),
+///     radius = 10,
+///     style = primitive_style!(stroke_color = Gray8::WHITE)
+/// );
+/// circle.draw(&mut display)?;
+///
+/// // Update the display
+/// display.flush().expect("Failed to send data to display");
+/// # Ok::<(), core::convert::Infallible>(())
+/// ```
+///
+/// ## Hardware Acceleration
+///
+/// In addition to defining [`draw_pixel`], an implementation of [`DrawTarget`] can also provide
+/// alternative implementations for hardware accelerated drawing operations. This example implements
+/// `DrawTarget` for a display that supports drawing hardware accelerated styled [`Rectangle`]s.
+///
+/// The `draw_rectangle()` method overrides the default implementation.
```

Comment:
IMO this sentence belongs together with the previous one in a single paragraph. The first one describes what is done and the second one how.